scalikejdbc-streams/src/main/scala/scalikejdbc/streams/ExtractedResultIterator.scala

```diff
@@ -0,0 +1,55 @@
+package scalikejdbc.streams
+
+import java.sql.ResultSet
+
+import scalikejdbc.{ ResultSetCursor, WrappedResultSet }
+
+abstract class ExtractedResultIterator[+A](rs: ResultSet, autoClose: Boolean)(extract: WrappedResultSet => A) extends BufferedIterator[A] with CloseableIterator[A] {
+  private[this] var state = 0 // 0: no data, 1: cached, 2: finished
+  private[this] var cached: A = null.asInstanceOf[A]
+
+  protected[this] final def finished(): A = {
```

Comment:
This method is not a predicate function and does two things (changing the state and returning null value). How about using more expressive name such as `markedAsFinishedAndReturnNullValue`?